content/general/functions.textile

```diff
@@ -8,17 +8,176 @@ Reactor Functions allow your server-less functions to be invoked following chann
 
-Generally, customers who want to receive events as they happen, use our "message queues":/general/queues or "firehose":/general/firehose, or alternatively our "realtime client libraries":/realtime and subscribe to events and messages published. However, some customers prefer a server-less architecture and want to have their code invoked when a user publishes a message, attaches to a channel, or enters the presence set on a channel. For example, a developer may want to send a welcome message to someone when they first enter a chat channel. A reactor Function can be used to achieve this.
+Generally, customers who want to receive events as they happen, use our "message queues":/general/queues or "firehose":/general/firehose, or alternatively our "realtime client libraries":/realtime and subscribe to events and messages published. However, some customers prefer a server-less architecture and want to have their code invoked when a user publishes a message, attaches to a channel, or enters the presence set on a channel. For example, a developer may want to send a welcome message to someone when they first enter a chat channel. A Reactor Function can be used to achieve this.
+
+Reactor Functions are are designed to trigger server-less functions on common platforms such as Amazon and Google.  If you are looking to trigger the execution of code on your own servers over HTTP when a message is published or event occurs, see our "Reactor WebHooks":/general/webhooks.
+
+h2. Message Format
+
+The format of messages sent can be defined in your "account dashboard":https://support.ably.io/solution/articles/3000048664-how-do-i-access-my-account-dashboard. You can have the message sent as a JSON message, or as a "MessagePack":https://msgpack.org.
+
+* JSON (JavaScript Object Notation): An efficient data-interchange format which is fairly standard and provides simple text based encoding.
+* MsgPack: An efficient binary serialization format that is similar to JSON, but faster and smaller.
+
+
+h2. Envelopes
+
+Messages sent by Reactor Functions can be contained within an envelope. This can be useful for clients which are unable to access response headers or status codes.
 
-Reactor Functions are are designed to trigger server-less functions on common platform such as Amazon and Google.  If you are looking to trigger execution of code on your own servers over HTTP when a message is published or event occurs, see our "Reactor WebHooks":/general/webhooks.
 
 h2. Payload types
 
-Ably currently supports three types of data that can be delivered via our Firehose:
+Ably currently supports two types of data that can be delivered via our Functions:
+
+* "Messages":#messages - messages trigger function calls as soon as they are published on a channel
+* "Presence events":#presence - when clients enter, update their data, or leave channels, the presence event triggers a function call
```

Comment:
Update presence events desc with `presence events trigger function calls when clients enter, update their data, or leave channels.` to match the tone with the messages desc ?